umm-spec-lib/src/cmr/umm_spec/spatial_conversion.clj

```diff
@@ -20,30 +21,37 @@
    "WRS-1"
    "WRS-2"])
 
+(def upcase-valid-tile-identification-system-names
+  "Upcase values to allow for case-insenitive validation"
+  (map util/safe-uppercase valid-tile-identification-system-names))
+
 (defn tile-id-system-name-is-valid?
   "Return whether or not the given TileIdentificationSystemName is one of the
    valid-tile-identification-system-names"
   [tile-system-id-name]
-  (some #(= tile-system-id-name %) valid-tile-identification-system-names))
+  (some #(= (util/safe-uppercase tile-system-id-name) %) upcase-valid-tile-identification-system-names))
 
 (defn translate-tile-id-system-name
   "Return nil or equivalent value if the given name does not match any
    in the list of valid ones"
   [tile-identification-system-name]
   (when (tile-id-system-name-is-valid? tile-identification-system-name)
-    tile-identification-system-name))
+    (util/match-enum-case tile-identification-system-name valid-tile-identification-system-names)))
 
 (defn expected-tiling-id-systems-name
   "Translate TilingIdentificationSystemNames in accordance with UMM Spec v1.10"
   [tiling-identification-systems]
-  (let [tiling-id-systems (mapv
-                           (fn [tiling-id-system]
-                             (update tiling-id-system
-                                     :TilingIdentificationSystemName
-                                     translate-tile-id-system-name))
-                           tiling-identification-systems)]
-    (when-not (empty? tiling-id-systems)
-      tiling-id-systems)))
+  (->> tiling-identification-systems
+       (mapv #(update % :TilingIdentificationSystemName translate-tile-id-system-name))
+       seq))
+
+(defn filter-and-translate-tiling-id-systems
+  "Drop invalid TilingIdentificationSystems, and ensure that valid ones are cased
+   properly."
+  [tiling-identification-systems]
+  (->> tiling-identification-systems
+       (filter #(tile-id-system-name-is-valid? (:TilingIdentificationSystemName %)))
+       expected-tiling-id-systems-name))
```

Comment:
You could simplify the code by combining the filter-and-translate-tiling-id-systems, expected-tiling-id-system-name and translate-title-id-system-name because now that you have the generic match-enum-case function, you don't need to check if the tiling-id is valid anymore.

Reply:
These functions are used in different places, so deleting them entirely would be a more significant effort. I'd prefer to avoid this if I can, but if you're adamant I'll refactor as needed

Reply:
It's really not a lot of changes. I checked the baseline code. There are no other places that reference these functions other than one or two places in this PR.  After you drop tile-id-system-name-is-valid? function, you will see there is no need for both filter-and-translate-tiling-id-systems and expected-tiling-id-systems-name - only one of them is needed because they are the same. Regarding translate-title-id-system-name, after tile-id-system-name-is-valid? is removed, there is really nothing more than just use the match-enum-case directly.  so if you want to keep translate-title-id-system-name it's fine. Same with the Vertical Spatial Domain case.